07-routing-gossip.md

```diff
@@ -208,84 +224,92 @@ nodes for which a channel is not already known are ignored.
+The creating node:
+  - MUST set `timestamp` to be greater than that of any previous
+  `node_announcement` it has previously created.
+    - MAY base it on a UNIX timestamp.
+  - MUST set `signature` to the signature of the double-SHA256 of the entire
+  remaining packet after `signature` (using the key given by `node_id`).
+  - MAY set `alias` AND `rgb_color` to customize its appearance in maps and
+  graphs.
+    - Note: the first byte of `rgb` is the red value, the second byte is the
+    green value, and the last byte is the blue value.
+  - MUST set `alias` to a valid UTF-8 string, with any `alias` trailing-bytes
+  equal to zero.
+  - SHOULD fill `addresses` with an address descriptor for each public network
+  address that expects incoming connections.
+  - MUST set `addrlen` to the number of bytes in `addresses`.
+  - MUST place non-zero typed address descriptors in ascending order.
+  - MAY place any number of zero-typed address descriptors anywhere.
+  - SHOULD use placement only for aligning fields that follow `addresses`.
+  - MUST NOT create a `type 1` OR `type 2` address descriptor with `port` equal
+  to zero.
+  - SHOULD ensure `ipv4_addr` AND `ipv6_addr` are routable addresses.
+  - MUST NOT include more than one `address descriptor` of the same type.
+  - SHOULD set `flen` to the minimum length required to hold the `features`
+  bits it sets.
+
+The receiving node:
+  - if `node_id` is NOT a valid compressed public key:
+    - SHOULD fail the connection.
+    - MUST NOT further process the message.
+  - if `signature` is NOT a valid signature (using `node_id` of the
+  double-SHA256 of the entire message following the `signature` field, including
+  unknown fields following `alias`):
+    - SHOULD fail the connection.
+    - MUST NOT further process the message.
+  - if `features` field contains unknown _even bits_:
+    - MUST NOT parse the remainder of the message.
+    - MAY discard the message altogether.
+  - MAY forward `node_announcement`s that contain an unknown `features` bit,
+  regardless of if it has parsed the announcement or not.
+  - SHOULD ignore the first `address descriptor` that does NOT match the types
+  defined above.
+  - if `addrlen` is insufficient to hold the address descriptors of the
+  known types:
+    - SHOULD fail the connection.
+  - if `port` is equal to zero:
+    - SHOULD ignore `ipv6_addr` OR `ipv4_addr`.
+  - if `node_id` is NOT previously known from a `channel_announcement` message,
+  OR if `timestamp` is NOT greater than the last-received `node_announcement`
+  from this `node_id`:
+    - SHOULD ignore the message.
+  - otherwise:
+    - if `timestamp` is greater than the last-received `node_announcement` from
+    this `node_id`:
+      - SHOULD queue the message for rebroadcasting.
+      - MAY choose NOT to queue messages longer than the minimum expected length.
+  - SHOULD NOT connect to a node which has an unknown, _even_ `features` bit set
```

Comment:
Should this be up in "if `features` field contains unknown _even bits_:" (e.g. line 288) as "SHOULD NOT connect to that node".

Reply:
Good question @rustyrussell?